### DISCUSSION TITLE
How to use Expr::cust_with_values with other binds?

### DISCUSSION THREAD

**Discussion by negezor** [SeaQL/sea-orm#987] [Q&A] [answered]
I'm simulating postgres arrays using JSON, but it turns out that I can't add binds very easily. I have code:
```rs
let role_ids = vec![1, 2, 3];

let mut query = user::Entity::find();

if !role_ids.is_empty() {
    let binds = (0..role_ids.len())
        .map(|i| format!("${}", i + 1))
        .collect::<Vec<_>>()
        .join(",");
    
    query = query.filter(
        Expr::col(user::Column::RoleIds)
            .contains(
                Expr::cust_with_values(&format!("array_to_json(ARRAY[{}]::bigint[])::jsonb", binds), role_ids),
            ),
    );
}

let res = query
    .limit(10)
    .all(connection)
    .await?;
```
Table:
```sql
CREATE TABLE "public"."users" (
    "id" int8 NOT NULL,
    "role_ids" jsonb NOT NULL DEFAULT '[]'::jsonb,
    PRIMARY KEY ("id")
);
```
Query:
```sql
SELECT
  "users"."id",
  "users"."role_ids"
FROM
  "users"
WHERE
  "role_ids" @> array_to_json(ARRAY[$1,$2,$3]::bigint[])::jsonb
LIMIT
  $1
```
Result: `bind message supplies 1 parameters, but prepared statement \"sqlx_s_2\" requires 2`.
If I do an offset for binds:
```diff
- .map(|i| format!("${}", i + 1))
+ .map(|i| format!("${}", i + 2))
```
Query:
```sql
SELECT
  "users"."id",
  "users"."role_ids"
FROM
  "users"
WHERE
  "role_ids" @> array_to_json(ARRAY[$2,$3,$4]::bigint[])::jsonb
LIMIT
  $1
```
Result: `Query Error: error returned from database: bind message supplies 1 parameters, but prepared statement \"sqlx_s_2\" requires 3`
Now I use this code as a workaround:
```rs
Expr::cust(&format!("'[{}]'::jsonb", role_ids.into_iter().map(|s| s.to_string()).join(","))),
```
Is this a bindings error? I found a similar discussion https://github.com/SeaQL/sea-orm/discussions/487#discussioncomment-2088744

Postgres: `14.5`
Rust: `1.63.0`
SeaORM: `0.9.2`
SeqQuery: `0.26.3`

**Comment by billy1624** [comment, marked as answer]
Hey @negezor, welcome and sorry for the delay. I don't think it's a bug. The query itself have no idea `$N` is being used in `Expr::cust`.
